src/middlewared/middlewared/plugins/system.py

```diff
@@ -246,6 +247,7 @@ async def __validate_fields(self, schema, data):
             Str('syslog_transport', enum=['UDP', 'TCP', 'TLS']),
             Int('syslog_tls_certificate', null=True),
             List('isolated_gpu_pci_ids', items=[Str('pci_id')]),
+            List('kernel_extra_options', items=[Str('extra_option')]),
```

Comment:
Why does it have to be an array?

Reply:
Makes the most sense to me IMO. Each entry being one option, instead of having one very long string array.

The opposite would also be true, why would it be a single string?

Reply:
Kernel options is a string, that's why I am suggesting.

How will this look like in UI? Multiple `<input type=text>` with `+` button? Why not a single textarea?

Reply:
No UI for this feature

Reply:
Strings will be easier to use in CLI too

Reply:
@sonicaj we also have to prohibit symbols like `\n` here, and handle `"` in generator (or prohibit it too)

Reply:
I think we can have it as a string because that's also how we pass auxiliary parameters to other services. So we will be consistent that way as well.